src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -51,21 +54,25 @@
     private StackPane statusbarPlaceholder;
 
     /**
-     * Creates a {@code MainWindow} with the given {@code Stage} and {@code Logic}.
+     * Creates a {@code MainWindow} with the given {@code Stage} and {@code LogicPerson}.
      */
-    public MainWindow(Stage primaryStage, LogicPerson logic) {
+    public MainWindow(Stage primaryStage, LogicMode logicMode, LogicPerson logicPerson) {
         super(FXML, primaryStage);
 
         // Set dependencies
         this.primaryStage = primaryStage;
-        this.logic = logic;
+        this.logicMode = logicMode;
+        this.logicPerson = logicPerson;
```

Comment:
As mentioned above, we can discuss this soon :D But, I think we don't need to make any changes for this iteration 👍  

For a reminder, maybe we can add a TODO comment too here?